app/docker/views/networks/create/createNetworkController.js

```diff
@@ -1,138 +1,195 @@
+        };
+        config.Scope = 'swarm';
+      }
+
+      function validateForm(accessControlData, isAdmin) {
+        $scope.state.formValidationError = '';
+        var error = '';
+        error = FormValidator.validateAccessControl(accessControlData, isAdmin);
+
+        if (error) {
+          $scope.state.formValidationError = error;
+          return false;
+        }
+        return true;
+      }
+
+      function createNetwork(networkConfiguration, userDetails, accessControlData, reload) {
+        $scope.state.actionInProgress = true;
+        NetworkService.create(networkConfiguration)
+          .then(function success(data) {
+            var networkIdentifier = data.Id;
+            var userId = userDetails.ID;
+            return ResourceControlService.applyResourceControl('network', networkIdentifier, userId, accessControlData, []);
+          })
+          .then(function success() {
+            Notifications.success('Network successfully created');
+            if (reload)
+              $state.go('docker.networks', {}, {
+                reload: true
+              });
+          })
+          .catch(function error(err) {
+            Notifications.error('Failure', err, 'An error occured during network creation');
+          })
+          .finally(function final() {
+            $scope.state.actionInProgress = false;
+          });
+      }
+
+      $scope.create = function () {
+        var networkConfiguration = prepareConfiguration();
+        var accessControlData = $scope.formValues.AccessControlData;
+        var userDetails = Authentication.getUserDetails();
+        var isAdmin = userDetails.role === 1;
+
+        if (!validateForm(accessControlData, isAdmin)) {
+          return;
+        }
+
+        if ($scope.config.Driver === 'macvlan') {
+          if ($scope.formValues.Macvlan.Scope === 'local') {
+            var selectedNodes = $scope.formValues.Macvlan.DatatableState.selectedItems;
+            selectedNodes.forEach(function (node, idx) {
+              var nodeName = node.Hostname;
+              HttpRequestHelper.setPortainerAgentTargetHeader(nodeName);
+
+              modifyNetworkConfigurationForMacvlanConfigOnly(networkConfiguration);
+              createNetwork(networkConfiguration, userDetails, accessControlData, idx === selectedNodes.length - 1 ? true : false);
+            });
+          } else if ($scope.formValues.Macvlan.Scope === 'swarm') {
+            var selectedNetworkConfig = $scope.formValues.Macvlan.SelectedNetworkConfig;
+            HttpRequestHelper.setPortainerAgentTargetHeader(selectedNetworkConfig.NodeName);
+
+            modifyNetworkConfigurationForMacvlanConfigFrom(networkConfiguration, selectedNetworkConfig);
+            createNetwork(networkConfiguration, userDetails, accessControlData, true);
+          }
+        } else {
+          var nodeName = $scope.formValues.NodeName;
+          HttpRequestHelper.setPortainerAgentTargetHeader(nodeName);
```

Comment:
I believe that the following code block can be located outside of the `if/else` block:

```javascript
HttpRequestHelper.setPortainerAgentTargetHeader(nodeName);
createNetwork(networkConfiguration, userDetails, accessControlData, true);
```

Nodename should be computed in each block of the if/else as well as the last parameter of the `createNetwork` function.